### ISSUE TITLE
fatal error : /usr/libexec/nomadbsd-setup commit returned with error code 1: "gpart autofill: no space left on device ; gpart add -t freebsd vtbd0 failed"

### ISSUE THREAD

**Issue by Marietto2008** [nomadbsd/NomadBSD#129]
Hello.

I've installed NomadBSD on top of my FreeBSD system (FreeBSD marietto 13.1-RELEASE-p3 FreeBSD 13.1-RELEASE-p3 GENERIC amd64) and then I've virtualized the image produced with bhyve,using the following parameters :

```
bhyve -S -c sockets=1,cores=2,threads=2 -m 4G -w -H -A \
-s 0,hostbridge \
-s 1,virtio-blk,/home/marietto/bhyve/Files/nomadbsd-131R-80dec9b9.amd64.ufs.img,bootindex=1 \
-s 10,virtio-net,tap1 \
-s 11,virtio-9p,sharename=/ \
-s 29,fbuf,tcp=0.0.0.0:5901,w=1600,h=950,wait \
-s 30,xhci,tablet \
-s 31,lpc \
-l bootrom,/usr/local/share/uefi-firmware/BHYVE_UEFI_CODE.fd \
vm0:1 < /dev/null & sleep 2 && vncviewer 0:1
```

Unfortunately I seen two problems :

1) my mouse does not work : it means that I can't use it within NomadBSD virtualized inside the bhyve / vnc window

2) its happened the error that you see below :

![Screenshot_2022-11-17_00-27-10](https://user-images.githubusercontent.com/755339/202317725-8fcccc56-18ce-4cb9-a5fd-2f987548abdb.jpg)

**Comment by mrclksr** [comment]
Hi  @Marietto2008,

the image needs some extra space to expand. When you write it to a storage device, the setup will use the remaining space. When using VirtualBox, you would run `VBoxManage modifyhd nomadbsd-x.y.z.vdi --resize 8000` after creating a VDI. In your case you can use `truncate`:

```
# truncate +8g nomadbsd-131R-80dec9b9.amd64.ufs.img
```
This will add 8G extra space to the image.

You can do the same with the ZFS image, albeit it has already completed the setup. Truncate the ZFS image, then boot the ZFS image and execute `zpool online -e nomadbsd_zroot`

Regarding the mouse problem:
Try to disable `xhci`

**Comment by Marietto2008** [comment]
I've truncated the ufs image with this command :

truncate -s +10g nomadbsd-131R-80dec9b9.amd64.ufs.img

and the error 2 is gone

I have removed this parameter from bhyve : -s 30,xhci,tablet \

and the mouse works great. There still be one error,that affects the ufs and the zfs version,the one you see below :

![Screenshot_2022-11-17_11-59-05](https://user-images.githubusercontent.com/755339/202429286-cc78b20e-41d8-4a8a-b59d-43f112b69274.jpg)

**Comment by mrclksr** [comment]
Excellent!

The error is due to a missing sound card.

**Comment by Marietto2008** [comment]
I have a lot of sound cards on my system. On the picture below you can see all the channels I have. The one that's working on my host system is "USB audio" :

![Screenshot_2022-11-17_12-24-48](https://user-images.githubusercontent.com/755339/202434175-8309eb6b-2e9c-4dac-ad19-65b866612bd1.jpg)

between the bhyve parameters I have added a new parameter that emulates the sound card and that it always worked :

-s 13,hda,play=/dev/dsp,rec=/dev/dsp \

but this time it does not work within NomadBSD :

![Screenshot_2022-11-17_12-23-26](https://user-images.githubusercontent.com/755339/202434403-d86764eb-562d-47ca-9f68-78484b1e463e.jpg)

I've opened a youtube video and I didn't hear any sound. In addition,the streaming is very very slow.

**Comment by Marietto2008** [comment]
I've almost fixed the sound bug,by adding this parameter :

```
hw.snd.default_unit=8

to the file /etc/sysctl.conf
```

now,when I open a youtube video,I can hear sounds. BUT it goes very slow and the sound seems to be broken. So,there is still something to fix.

**Comment by bsdlme** [comment]
I'm not sure if bhyve is good enough to watch Youtube or maybe you need some additional parameters to bhyve?

**Comment by Marietto2008** [comment]
yeah,maybe you are right. using the framebuffer could not be enough to have a decent video speed. Maybe I can passthru my nvidia 2080 ti graphic card from the host to the nomadOS vm and this will improve the speed. I will try.
